src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -582,12 +582,12 @@ export class PersistentTerminalProcess extends Disposable {
 
 	async attach(): Promise<void> {
 		this._logService.trace('persistentTerminalProcess#attach', this._persistentProcessId);
+		// Something wrong happened if the disconnect runner is not canceled, this likely means
+		// multiple windows attempted to attach.
+		if (!await this._isOrphaned()) {
```

Comment:
This should never happen but I'm sure still does when the issue reproduces. This change means a new terminal will be created with the default profile instead of reconnecting.